File: school/mr/hw/april/5/25.cpp
#include <iostream>
#include <cmath>

int main() 
{
    for (int i = 101000000; i < 102000000; ++i)
    {
        int c = 0;
        for (int div = 1; div < int(std::sqrt(i)) + 1; ++div)
        {
            if (i % div != 0) continue;
            int div2 = i / div;
            if (div % 2 == 0) ++c;
            if (div2 % 2 == 0 && div2 != div) ++c;
            if (c > 3) break;
        }
        if (c == 3)
            std::cout << i << std::endl;
    }
    return 0;
}

File: school/mr/hw/april/5/25.py
for i in range(101_000_000, 102_000_000):
    c = 0

    for div in range(1, int(i ** 0.5) + 1):
        if i % div == 0:
            c += div % 2 == 0
            c += (i // div) % 2 == 0 and (i // div) != div
        if c > 3:
            c += 123
            break
    if c == 3:
        print(i)
